server.go

```diff
@@ -340,6 +392,173 @@ func (a *App) VerifyHandler(w http.ResponseWriter, r *http.Request) {
+	if err := dec.Decode(reqBody); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		enc.Encode(struct {
+			Message string `json:"message"`
+		}{"json decode error: " + err.Error()})
+		return
+	}
+
+	ll := &VerifiableLockList{}
+	locks, nextCursor, err := a.metaStore.FilteredLocks(auth, repo, "",
+		reqBody.Cursor,
+		strconv.Itoa(reqBody.Limit))
+	if err != nil {
+		ll.Message = err.Error()
+	} else {
+		ll.NextCursor = nextCursor
+
+		for _, l := range locks {
+			if strings.Contains(l.Path, "theirs") {
+				ll.Theirs = append(ll.Theirs, l)
+			} else {
+				ll.Ours = append(ll.Ours, l)
+			}
+		}
+	}
+
+	enc.Encode(ll)
+
+	logRequest(r, 200)
+}
+
+func (a *App) CreateLockHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	user := vars["user"]
+	repo := vars["repo"]
+	auth := r.Header.Get("Authorization")
+
+	dec := json.NewDecoder(r.Body)
+	enc := json.NewEncoder(w)
+
+	w.Header().Set("Content-Type", metaMediaType)
+
+	var lockRequest LockRequest
+	if err := dec.Decode(&lockRequest); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		enc.Encode(&LockResponse{Message: err.Error()})
+		return
+	}
+
+	locks, err := a.metaStore.Locks(auth, repo)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		enc.Encode(&LockResponse{Message: err.Error()})
+		return
+	}
+
+	for _, l := range locks {
+		if l.Path == lockRequest.Path {
+			w.WriteHeader(http.StatusConflict)
+			enc.Encode(&LockResponse{Message: "lock already created"})
+			return
+		}
+	}
+
+	var id [20]byte
+	rand.Read(id[:])
+
+	lock := &Lock{
+		Id:       fmt.Sprintf("%x", id[:]),
+		Path:     lockRequest.Path,
+		Owner:    User{Name: user},
+		LockedAt: time.Now(),
+	}
+
+	if err := a.metaStore.AddLocks(auth, repo, *lock); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		enc.Encode(&LockResponse{Message: err.Error()})
+		return
+	}
+
+	w.WriteHeader(http.StatusCreated)
+	enc.Encode(&LockResponse{
+		Lock: lock,
+	})
+
+	logRequest(r, 200)
+}
+
+func (a *App) DeleteLockHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	repo := vars["repo"]
+	lockId := vars["id"]
+	auth := r.Header.Get("Authorization")
+
+	dec := json.NewDecoder(r.Body)
+	enc := json.NewEncoder(w)
+
+	w.Header().Set("Content-Type", metaMediaType)
+
+	var unlockRequest UnlockRequest
+
+	if len(lockId) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		enc.Encode(&UnlockResponse{Message: "Invalid lock"})
+		return
+	}
+
+	if err := dec.Decode(&unlockRequest); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		enc.Encode(&UnlockResponse{Message: err.Error()})
+		return
+	}
+
+	l, err := a.metaStore.DeleteLock(auth, repo, lockId)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		enc.Encode(&UnlockResponse{Message: err.Error()})
+		return
+	}
+
+	if l != nil {
+		enc.Encode(&UnlockResponse{Lock: l})
+	} else {
+		enc.Encode(&UnlockResponse{Message: "unable to find lock"})
```

Comment:
Does this case result in a 200? If not, should L559 be moved to have different status codes within each of these branches?